MVVMIOS/ViewModel/MovieViewModel.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  MovieViewModel.swift
+//  MVVMIOS
+//
+//  Created by Ravi Bastola on 3/6/22.
+//
+
+import Foundation
+
+protocol MovieViewModelResultPresentable: AnyObject {
+    func presentResult(_ result: Result<[MovieResult], Error>?)
```

Comment:
I would send a `MovieViewState` instead of a `Result` so that the VC doesn't have to do any logic other than presentation. As it is now the VC will have to check for different types of errors, decide what to present for those, etc. 

Generally I have something like this:
```swift
enum MovieViewState {
  case error(MovieErrorViewState)
  case loading(MovieLoadingViewState) // for before the network has finished
  case loaded(MovieLoadedViewState)
}

struct MovieErrorViewState {
  let errorText...
  let errorColor...
  ...
}
struct MovieLoadingViewState {
  let loadingText
  ...
} 
struct MovieLoadedViewState {
  let movieTitle...
  let movieDescription...
  ...
}
```